packages/DateInput/src/DateInput.js

```diff
@@ -0,0 +1,186 @@
+import React from "react";
+import PropTypes from "prop-types";
+import moment from "moment";
+import CalendarIcon from "@paprika/icon/lib/Calendar";
+import Input from "@paprika/input";
+import { isMomentObjectOrNull } from "@paprika/helpers/lib/customPropTypes";
+import useI18n from "@paprika/l10n/lib/useI18n";
+
+const INPUT_PARSE_ERROR = "INPUT_PARSE";
+
+const propTypes = {
+  /** If the value of <input> is valid or not. */
+  hasError: PropTypes.bool,
+
+  /** Date format used while entering and parsing user input. */
+  dateFormat: PropTypes.string,
+
+  /** Selected date in moment object. */
+  date: isMomentObjectOrNull,
+
+  /** Possible date in moment object. This prop is used when user is typing date and we can't determinate exact date. */
+  possibleDate: isMomentObjectOrNull,
+
+  /** Date format used while displaying date. It should be human-friendly and spelled out, default is MMMM DD,YYYY */
+  humanFormat: PropTypes.string,
+
+  /** Callback when date is inputed. Will be called on blur or enter key press. */
+  onChange: PropTypes.func,
+
+  /** Callback when user inputs date. Will be called after every key up event. */
+  onChangePossibleDate: PropTypes.func,
+
+  /** Error callback. Will be called on blur or enter key press if inputted date can't be parsed. */
+  onError: PropTypes.func,
+
+  onClick: PropTypes.func,
+
+  /** Guard function. If it returns true - confirmation will be prevented. */
+  denyConfirmation: PropTypes.func,
```

Comment:
out of curiosity in what scenario we would be using this prop?

Reply:
@nahumzs 

"confirmation" in context of DateInput - is a moment when inputted date is converting from `dateFormat` to `humanFormat` and `onChange` callback is calling. Normally "confirmation" occurs on DateInput blur. But there is specific scenario:
1. we use DateInput inside DatePicker
2. we click in DateInput and calendar should be shown
3. If we click somewhere on blank space Calendar should be hidden and "confirmation" should be triggered. But if we click on shown Calendar, we should NOT trigger "confirmation".

But we have problem: DateInput does NOT know anything about shown Calendar. So I decided to pass this function from DatePicker to DateInput, to prevent "confirmation".

I am 100% sure that this is not ideal solution. Especially because we should call passed down `denyConfirmation` function via `requestAnimationFrame`. It is required to delay resolving `denyConfirmation`, because input blur handler will be called before calendar receive a focus. By using `requestAnimationFrame` we check should "confirmation" be called or not only after calendar will receive focus.